tasks/kmt.py

```diff
@@ -281,43 +362,50 @@ def prepare(ctx, vms, stack=None, arch=None, ssh_key=None, rebuild_deps=False, p
     if not arch:
         arch = platform.machine()
 
-    if not compiler_running(ctx):
-        start_compiler(ctx)
-
-    download_gotestsum(ctx)
-
     infra = build_infrastructure(stack, ssh_key)
     domains = filter_target_domains(vms, infra)
+    build_from_scratch = (
+        full_rebuild
+        or (not os.path.exists(f"kmt-deps/{stack}"))
+        or (not vms_have_correct_deps(ctx, domains, os.path.join("kmt-deps", stack, f"dependencies-{arch}.tar.gz")))
+    )
+
+    if not compiler_running(ctx):
+        start_compiler(ctx)
 
     constrain_pkgs = ""
-    if not rebuild_deps:
+    if not build_from_scratch:
         constrain_pkgs = f"--packages={packages}"
 
     docker_exec(
         ctx,
         f"git config --global --add safe.directory /datadog-agent && inv -e system-probe.kitchen-prepare --ci {constrain_pkgs}",
         run_dir="/datadog-agent",
     )
-    if rebuild_deps:
-        docker_exec(
-            ctx,
-            f"./test/new-e2e/system-probe/test/setup-microvm-deps.sh {stack} {os.getuid()} {os.getgid()} {platform.machine()}",
-            run_dir="/datadog-agent",
+
+    target_instances = list()
+    for d in domains:
```

Comment:
Do we want to validate that all instances share the same architecture as the one defined by `arch`? In my PR I do call this function using filtered vms, but even in that case someone might call it incorrectly from CLI.

Reply:
For now I have disallowed any cross-arch test/builds. We can revisit this once we have support for cross-compilation in KMT.
https://github.com/DataDog/datadog-agent/pull/22967/commits/d2c040076e5933cb19162b25c53c3b0caef90381